### DISCUSSION TITLE
Start workflow from DefinitionsList

### DISCUSSION THREAD

**Discussion by bskfrederik** [elsa-workflows/elsa-core#4510] [Ideas]
Hello,

we are currently integrating Elsa V3 into our product. We have use case that workflows should be started via UI. I have built a small prototype to demonstrate our desired behavior. When you click on the menu a button appears to trigger the workflow. 

![start_workflow](https://github.com/elsa-workflows/elsa-core/assets/131231567/015fd433-3d0a-4b22-9156-66736c2f1099)

There are currently two approaches i can think of to implement this feature: 

1. A "hardcoded" start button
2. A generic approach to register a button and a callback function to support custom actions in the menu

@sfmskywalker what do you think of the idea? I would appreciate your feedback :)

**Comment by sfmskywalker** [comment]
I love it, and I think you should go ahead and hardcode that button in. After all, the designer also hardcodes this button in the toolbar. Ideally, both your new button and the existing button invoke the same code - which might mean some refactoring needs to happen.
The benefit of that is that when we implement a dialog to allow the user to override default inputs, both buttons will invoke that dialog automatically.

**Reply by bskfrederik** [reply]
Hi, thanks for your feedback. I've been looking for the button you mentioned but haven't found it in the designer. Where can I find this button @sfmskywalker?

**Reply by bskfrederik** [reply]
Hi, I have found the button. It is currently commented out in `toolbar.tsx` in elsa-workflows-designer.

**Reply by sfmskywalker** [reply]
Great! Please keep in mind that we’re retiring the Stencil implementation and instead are evolving the Blazor implementation.
